src/app/ui/components/item/item.tsx

```diff
@@ -0,0 +1,45 @@
+import { forwardRef } from 'react';
+
+import { css } from 'leather-styles/css';
+import { styled } from 'leather-styles/jsx';
+import { type ItemVariantProps, item as itemRecipe } from 'leather-styles/recipes';
+import { HTMLStyledProps } from 'leather-styles/types';
+
+import { HasChildren } from '@app/common/has-children';
+
+export const itemCaptionStyles = css({
+  _groupDisabled: { color: 'accent.non-interactive' },
+  color: 'accent.text-subdued',
+});
+
+export const itemChevronStyles = css({
+  _groupDisabled: { color: 'accent.non-interactive' },
+  color: 'accent.action-primary-default',
+});
+
+const ItemBase = forwardRef<HTMLDivElement, HTMLStyledProps<'div'> & ItemVariantProps>(
+  (props, ref) => <styled.div className={`group ${itemRecipe()}`} ref={ref} {...props} />
+);
+
+const ItemInteractive = forwardRef<HTMLButtonElement, HTMLStyledProps<'button'> & ItemVariantProps>(
+  (props, ref) => (
+    <styled.button className={`group ${itemRecipe({ interactive: true })}`} ref={ref} {...props} />
+  )
+);
+
+interface RootProps {
+  isDisabled?: boolean;
+  isInteractive?: boolean;
+  onClickItem?(): void;
+}
+function Root({ children, isDisabled, isInteractive, onClickItem }: RootProps & HasChildren) {
+  return isInteractive ? (
+    <ItemInteractive disabled={isDisabled} onClick={onClickItem}>
+      {children}
+    </ItemInteractive>
+  ) : (
+    <ItemBase>{children}</ItemBase>
+  );
+}
```

Comment:
There's no value in using a slot component if we just had a single one. Recommend changing to a single component.

On `isInteractive`, this could just be inferred from whether there's a `onClick` defined or not? This redundancy creates the possibility for illogical props `isInteractive={false} onClick={() => {...}}`

I reckon that, as the behaviours of `ItemInteractive` and `ItemBase` are different, they don't belong tied together in a single component. It's only because of past use of `isPressable` that they are in this solution. If this is too complex to remove, I think it's okay to leave as this, I just question whether this ternary logic should belong in the component itself.

For instance, what if we were to remove `isPressable`, and in all the places where it's used we just use `ItemInteractive` and disable it if there isn't an action? That would make more sense I think?

Reply:
I'm not using a slot component, but did use the naming of `Root` so that is prob confusing. I am just using a config recipe, not a slot recipe in this refactor.

I am using `isDefined(onClick)` in the component to set the `isInteractive` prop, but I can move to handle it in the item, np ...just thought it was more explicit with our use atm.

We can't really remove the `isPressable` situation w/out a bigger refactor here, and I don't think we should take the time to do it rn with new designs coming anyway for most of these features? I thought about, and tried, just disabling the `ItemInteractive` when not pressable, not a disabled state/styling is really different than this. That is why I chose to use the share base styling and applying the variant of `interactive` to a different component. Also, we do get the console error of a button in a button in some cases (with the Connect Ledger button) when it is embedded in the asset list.